### ISSUE TITLE
[Bug]: Move Fakes to a test module

### ISSUE THREAD

**Issue by JoseAlcerreca** [android/nowinandroid#875]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is there a StackOverflow question about this issue?

- [X] I have searched StackOverflow

### What happened?

`core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/fake` has a bunch of fakes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SimonMarquis** [comment]
> So, basically we have to move these bunch of repositories to core/data/src/test/java/com/google/samples/apps/nowinandroid/core/data/repository? Isn't?

No, because we probably need to use them in other modules, and you can't depend on test sourceSets of other modules.

FWIW, I think that until we have proper AGP support for testFixtures in Kotlin for Android modules, migrating fakes to a centralized module is a bad practice in terms of modularisation. It encourages devs to merge unrelated code together and/or to create "test" modules for every other module.

To start this migration, one step forward would be to make this `:core:data` a pure JVM module. Only `ConnectivityManagerNetworkMonitor` brings the "Android" dependency.

For reference:
- https://github.com/android/nowinandroid/issues/451

**Comment by JoseAlcerreca** [comment]
> to create "test" modules for every other module

Why is this a bad thing? We could create a `:test` category if it becomes annoying.

**Comment by SimonMarquis** [comment]
This might not have any meaningful impact on small-scale projects, but if you have tens (or hundreds) of modules, and each of them has an extra companion module, this creates unnecessary load on the build process, while `testFixtures` has been designed specifically to solve this issue.
This will also break many annotations that rely on detecting the test classpath like `@VisibleForTesting`, `@TestOnly`, etc., and other similar features like IDE scopes, Lint, etc.
So, I'm not sure we should make this "companion module" approach a general solution to this problem.

**Comment by SimonMarquis** [comment]
Ping @JoseAlcerreca the fakes have been moved to the `:core:data-test` in this PR (now that they are no longer used outside):
- #1144
